### DISCUSSION TITLE
[FeatureRequest] Program alarms etc through light sensor

### DISCUSSION THREAD

**Discussion by krekr** [joeycastillo/Sensor-Watch#60] [Ideas]
I've been following this project for some time and recently got a hold of a A158WA, the metal sibling of the F91. It occurred to me that most 'sensors' would be very gimmicky in nature (temp, accel, etc.). No offence. But if you could get a light sensor (LDR or similar) in there, and could get light to it when mounted inside the watch, you'd have one-way communication possibilities.

The protocol could be similar to the provisioning sequence that Electric Imp (IoT devices in SD-card form factor) uses ("BlinkUp") https://developer.electricimp.com/platform-overview

This would open up a host of interesting possibilities; programming alarms through a nice GUI on a smartphone or laptop, sync with calendars, upload new info for 2FA etc.
I could design the board, but the firmware is beyond me. I'd be interested to know if others see this as useful as well.

**Comment by tandy-1000** [comment]
This sounds like a really useful feature!
I write Nim code and would be able to help add firmware support for this :)

**Reply by krekr** [reply]
Cool. I could do the hardware and send a version to you. This would be quite low-prio for me, but I'll let you know how it goes.
Any other people interested in this as well?
Do you already have a sensorwatch btw?

**Reply by tandy-1000** [reply]
Awesome, that's fine, I don't actually have a sensor watch yet so that will give me some time to procure one :).

**Comment by bademux** [comment]
Well, it can be acceptable tradeoff for https://github.com/joeycastillo/Sensor-Watch/discussions/52

There are great [news](https://twitter.com/josecastillo/status/1524833415756906496)  and looks like I need to learn how to obtain my light sensor for TOTP config. 
Or maybe receive files over irda for new [FS](https://twitter.com/josecastillo/status/1524018740521840640 ).
Is it possible to put IR receiver\transmitter? @joeycastillo  what do you think.
